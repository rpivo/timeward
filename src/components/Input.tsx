import React, { forwardRef } from 'react';

export type InputProps = {
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  password?: boolean;
  placeholder?: string;
  value?: string;
};

const Input = forwardRef(
  ({ onChange, password, placeholder, value }: InputProps, ref?: React.Ref<HTMLInputElement>) => {
    Input.displayName = 'Input';
    return (
      <input
        onChange={onChange}
        autoComplete={password ? 'on' : undefined}
        placeholder={placeholder}
        type={password ? 'password' : 'text'}
        ref={ref}
        value={value}
      />
    );
  },
);


export default Input;
