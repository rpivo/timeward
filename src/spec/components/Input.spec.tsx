import React from 'react';
import renderer from 'react-test-renderer';
import Input from '@components/Input';

describe('Input', () => {

  describe('render', () => {
    const testRenderer = renderer.create(<Input />);

    it('should render correctly', () => {
      testRenderer.toJSON();
      expect(testRenderer).toMatchSnapshot();
    });

  });
});
