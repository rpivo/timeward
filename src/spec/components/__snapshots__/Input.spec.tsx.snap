// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render should render correctly 1`] = `
Array [
  <label
    htmlFor=""
    style={
      Object {
        "display": "none",
      }
    }
  />,
  <input
    type="text"
  />,
]
`;
